easybuild/easyblocks/g/gromacs.py

```diff
@@ -439,49 +478,68 @@ def sanity_check_step(self):
+        mpi_bins = []
+        mpi_libnames = []
         if LooseVersion(self.version) < LooseVersion('5.1'):
-            bins.extend(['editconf', 'g_lie', 'genbox', 'genconf', 'mdrun'])
+            mpi_bins.extend(['mdrun'])
 
         if LooseVersion(self.version) >= LooseVersion('5.0'):
-            bins.append('gmx')
-            libnames.append('gromacs')
-            if LooseVersion(self.version) < LooseVersion('5.1') and self.toolchain.options.get('usempi', None):
-                bins.append('mdrun')
+            mpi_bins.append('gmx')
+            mpi_libnames.append('gromacs')
         else:
-            libnames.extend(['gmxana', 'gmx', 'md'])
-            # note: gmxpreprocess may also already be there for earlier versions
-            if LooseVersion(self.version) > LooseVersion('4.6'):
-                libnames.append('gmxpreprocess')
+            bins.extend(['editconf', 'g_lie', 'genbox', 'genconf'])
+            libnames.extend(['gmxana'])
+            if LooseVersion(self.version) >= LooseVersion('4.6'):
+                if self.dynamic:
+                    mpi_libnames.extend(['gmx', 'md'])
+                else:
+                    libnames.extend(['gmx', 'md'])
+            else:
+                mpi_libnames.extend(['gmx', 'md'])
+
+            if LooseVersion(self.version) >= LooseVersion('4.5'):
+                if LooseVersion(self.version) >= LooseVersion('4.6'):
+                    if self.dynamic:
+                        mpi_libnames.append('gmxpreprocess')
+                    else:
+                        libnames.append('gmxpreprocess')
+                else:
+                    mpi_libnames.append('gmxpreprocess')
 
         # also check for MPI-specific binaries/libraries
         if self.toolchain.options.get('usempi', None):
             if LooseVersion(self.version) < LooseVersion('4.6'):
-                mpisuff = self.cfg['mpisuffix']
+                mpisuff = self.cfg.get('mpisuffix')
             else:
                 mpisuff = '_mpi'
 
-            bins.extend([binary + mpisuff for binary in bins])
-            libnames.extend([libname + mpisuff for libname in libnames])
+            mpi_bins.extend([binary + mpisuff for binary in mpi_bins])
+            mpi_libnames.extend([libname + mpisuff for libname in mpi_libnames])
 
         suff = ''
 
         # make sure that configopts is a list:
-        configopts_list = self.cfg['configopts']
+        configopts_list = self.cfg.get('configopts')
         if isinstance(configopts_list, str):
             configopts_list = [configopts_list]
 
         lib_files = []
         bin_files = []
 
-        for configopts in configopts_list:
-            # add the _d suffix to the suffix, in case of the double precission
-            if re.search('DGMX_DOUBLE=(ON|YES|TRUE|Y|[1-9])', configopts, re.I):
-                suff = '_d'
+        if self.cfg.get('double_precision') is None or self.cfg.get('double_precision'):
```

Comment:
I think this change is wrong:

1. `double_precision` seems to only be used to set configopts and print messages, so this `if` can be removed which helps discussion in https://github.com/easybuilders/easybuild-easyblocks/pull/1991/files#r404754159
2. You moved the `lib_files.extend` out of the loop hence you only check for double precision libs.
3. The code before was wrong anyway and worked only if single precision came before double precision. `suff = ''` has to be in the loop